mifospay/src/main/java/org/mifospay/history/specific_transactions/presenter/SpecificTransactionsViewModel.kt

```diff
@@ -0,0 +1,85 @@
+package org.mifospay.history.specific_transactions.presenter
+
+import android.util.Log
+import androidx.compose.runtime.MutableState
+import androidx.lifecycle.ViewModel
+import com.mifospay.core.model.domain.Transaction
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import org.mifospay.R
+import org.mifospay.core.data.base.TaskLooper
+import org.mifospay.core.data.base.UseCase
+import org.mifospay.core.data.base.UseCaseFactory
+import org.mifospay.core.data.domain.usecase.account.FetchAccountTransfer
+import org.mifospay.core.data.util.Constants
+import javax.inject.Inject
+
+@HiltViewModel
+class SpecificTransactionsViewModel @Inject constructor(
+    private val mUseCaseFactory: UseCaseFactory,
+    private var mTaskLooper: TaskLooper? = null
+): ViewModel() {
+
+    private val _uiState: MutableStateFlow<SpecificTransactionsUiState> = MutableStateFlow(SpecificTransactionsUiState.Loading)
+    val uiState get() = _uiState
+
+    fun getSpecificTransactions(
+        transactions: ArrayList<Transaction>?,
```

Comment:
Using nullable in transactions makes no sense

Reply:
done